### ISSUE TITLE
isolate query deduplication in a task

### ISSUE THREAD

**Issue by Geal** [apollographql/router#2773]
similar to how timeouts in query planning deduplication were fixed, in query deduplication (in traffic shaping), if the client request that triggered the initial deduplicated query is canceled, then it will fail for all the other requests that were waiting on it. We should instead start that execution in a task (but still under the same trace), and get it to completion even if the parent request was canceled, so that other requests still get the result

**Comment by o0Ignition0o** [comment]
@Geal I think we implemented it?

**Comment by Geal** [comment]
we did not. Still something to look at, now that query deduplication is used more frequently, but not a priority right now I guess
